homeassistant/components/brother/config_flow.py

```diff
@@ -64,6 +69,61 @@ class BrotherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_zeroconf(self, user_input=None):
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        try:
+            if not user_input.get("name").startswith("Brother"):
```

Comment:
This will error if the dict lookup returns `None`. Please use `dict[key]` if we know that the key is in the dict.